app/controllers/application_controller.rb

```diff
@@ -2,4 +2,32 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :logged_in?, except: [:index, :show]
```

Comment:
I'd rename logged_in? to something like ensure_logged_in or ensure_current_user - having a method called logged_in? suggests strongly that it will return a boolean and have no side effects.